src/script.rs

```diff
@@ -403,6 +392,72 @@ impl ScriptNum {
     }
 }
 
+impl From<i64> for ScriptNum {
+    fn from(value: i64) -> Self {
+        ScriptNum(value)
+    }
+}
+
+impl From<i32> for ScriptNum {
+    fn from(value: i32) -> Self {
+        ScriptNum(value.into())
+    }
+}
+
+impl From<u8> for ScriptNum {
+    fn from(value: u8) -> Self {
+        ScriptNum(value.into())
+    }
+}
+
+/// TODO: This instance will be obsolete if we convert bool directly to a `Vec<u8>`, which is also
+///       more efficient.
+impl From<bool> for ScriptNum {
+    fn from(value: bool) -> Self {
+        ScriptNum(value.into())
+    }
+}
+
+impl TryFrom<usize> for ScriptNum {
+    type Error = TryFromIntError;
+    fn try_from(value: usize) -> Result<Self, Self::Error> {
+        value.try_into().map(ScriptNum)
+    }
+}
+
+/// This is the primary extractor for numbers from `Script`, all other extractors should call this.
+impl From<ScriptNum> for i32 {
+    fn from(value: ScriptNum) -> Self {
+        if value.0 > i32::MAX.into() {
+            i32::MAX
+        } else if value.0 < i32::MIN.into() {
+            i32::MIN
+        } else {
+            value.0.try_into().unwrap()
+        }
+    }
+}
+
+impl TryFrom<ScriptNum> for u16 {
+    type Error = TryFromIntError;
+    fn try_from(value: ScriptNum) -> Result<Self, Self::Error> {
+        i32::from(value).try_into()
+    }
+}
+
+impl TryFrom<ScriptNum> for u8 {
+    type Error = TryFromIntError;
+    fn try_from(value: ScriptNum) -> Result<Self, Self::Error> {
+        i32::from(value).try_into()
+    }
+}
+
+impl From<ScriptNum> for Vec<u8> {
+    fn from(value: ScriptNum) -> Self {
+        ScriptNum::serialize(&value.0)
+    }
+}
+
```

Comment:
This is another that I don't think should be a `From`. `ScriptNum::serialize` is adequate.